File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/api/UserApi.java
package com.lattis.ellipse.data.network.api;

import com.lattis.ellipse.data.network.model.body.authentication.AcceptTermAndConditionBody;
import com.lattis.ellipse.data.network.model.body.commun.VerificationCodeBody;
import com.lattis.ellipse.data.network.model.body.user.ChangePasswordBody;
import com.lattis.ellipse.data.network.model.body.user.GetUserCurrentStatusBody;
import com.lattis.ellipse.data.network.model.body.user.SendCodeUpdatePhoneNumberBody;
import com.lattis.ellipse.data.network.model.body.user.SendForgotPasswordCodeBody;
import com.lattis.ellipse.data.network.model.body.user.UpdateEmailCodeBody;
import com.lattis.ellipse.data.network.model.body.user.UpdateUserBody;
import com.lattis.ellipse.data.network.model.body.user.ValidateCodeForChangePhoneNumberBody;
import com.lattis.ellipse.data.network.model.body.user.ValidateCodeUpdateEmailBody;
import com.lattis.ellipse.data.network.model.response.AddPrivateNetworkResponse;
import com.lattis.ellipse.data.network.model.response.BasicResponse;
import com.lattis.ellipse.data.network.model.response.CheckTermsConditionResponse;
import com.lattis.ellipse.data.network.model.response.ConfirmCodeForForgotPasswordBody;
import com.lattis.ellipse.data.network.model.response.GetCurrentUserStatusResponse;
import com.lattis.ellipse.data.network.model.response.GetTermsAndConditionsResponse;
import com.lattis.ellipse.data.network.model.response.GetUserResponse;

import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import io.reactivex.Observable;

public interface UserApi {

    @POST("users/get-user/")
    Observable<GetUserResponse> getUser();

    @POST("users/update-user/")
    Observable<GetUserResponse> saveUser(@Body UpdateUserBody updateUserBody);


    @POST("users/update-phone-number/")
    Observable<Void> validateCodeForChangePhoneNumber(@Body ValidateCodeForChangePhoneNumberBody body);

    @POST("users/update-phone-number-code/")
    Observable<Void> sendCodeForPhoneNumber(@Body SendCodeUpdatePhoneNumberBody body);

    @GET("users/update-password-code/")
    Observable<Void> sendCodeToUpdatePassword();


    @POST("users/change-password/")
    Observable<Void> validateCodeForChangePassword(@Body ChangePasswordBody body);

    @POST("users/get-current-status/")
    Observable<GetCurrentUserStatusResponse> getCurrentUserStatus(@Body GetUserCurrentStatusBody getUserCurrentStatusBody);

    @POST("users/update-email-code/")
    Observable<Void> sendCodeForUpdateEmail(@Body UpdateEmailCodeBody updateEmailCodeBody);

    @POST("users/add-private-account/")
    Observable<AddPrivateNetworkResponse> addPrivateNetworkEmail(@Body UpdateEmailCodeBody updateEmailCodeBody);


    @GET("users/delete-account/")
    Observable<BasicResponse> deleteUserAccount();

    @GET("users/terms-and-conditions/")
    Observable<GetTermsAndConditionsResponse> getTermsAndCondition();

    @POST("users/accept-terms-and-conditions/")
    Observable<BasicResponse> acceptTermsAndCondition(@Body AcceptTermAndConditionBody acceptTermAndConditionBody);

    @GET("users/check-accepted-terms-and-conditions/")
    Observable<CheckTermsConditionResponse> checkTermsAndCondition();

    @POST("users/forgot-password")
    Observable<BasicResponse> sendForgotPasswordCode(@Body SendForgotPasswordCodeBody sendForgotPasswordCodeBody);

    @POST("users/confirm-forgot-password/")
    Observable<BasicResponse> confirmCodeForForgotPassword(@Body ConfirmCodeForForgotPasswordBody confirmCodeForForgotPasswordBody);

    @POST("users/update-email/")
    Observable<Void> validateCodeForChangeMail(@Body ValidateCodeUpdateEmailBody validateCodeUpdateEmailBody);

    @POST("users/confirm-email-verification-code/")
    Observable<Void> confirmVerificationCodeForPrivateNetwork(@Body VerificationCodeBody verificationCodeBody);
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/repository/datasources/api/user/UserApi.kt
package com.lattis.data.repository.datasources.api.user

import com.lattis.data.entity.body.authentication.VerificationCodeBody
import com.lattis.data.entity.body.user.*
import com.lattis.data.entity.response.BasicResponse
import com.lattis.data.entity.response.user.AddPrivateNetworkResponse
import com.lattis.data.entity.response.user.GetUserFleetsResponse
import com.lattis.data.entity.response.user.GetUserResponse
import com.lattis.data.entity.response.user.current_status.GetCurrentUserStatusResponse
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT

interface UserApi{

    @POST("api/users/get-current-status/")
    fun getCurrentUserStatus(@Body getUserCurrentStatusBody: GetUserCurrentStatusBody): Observable<GetCurrentUserStatusResponse>

    @get:POST("api/users/get-user/")
    val user: Observable<GetUserResponse>

    @POST("api/users/update-user/")
    fun saveUser(@Body updateUserBody: UpdateUserBody): Observable<GetUserResponse>

    @POST("api/users/confirm-email-verification-code/")
    fun confirmVerificationCodeForPrivateNetwork(@Body verificationCodeBody: VerificationCodeBody): Observable<BasicResponse>

    @POST("api/users/add-private-account/")
    fun addPrivateNetworkEmail(@Body updateEmailCodeBody: UpdateEmailCodeBody): Observable<AddPrivateNetworkResponse>

    @POST("api/users/update-phone-number-code/")
    fun sendCodeForPhoneNumber(@Body body: SendCodeUpdatePhoneNumberBody): Observable<BasicResponse>

    @POST("api/users/update-phone-number/")
    fun validateCodeForChangePhoneNumber(@Body body: ValidateCodeForChangePhoneNumberBody): Observable<BasicResponse>

    @POST("api/users/change-password/")
    fun changePassword(@Body body: ChangePasswordBody): Observable<BasicResponse>

    @POST("api/users/update-email/")
    fun validateCodeForChangeMail(@Body validateCodeUpdateEmailBody: ValidateCodeUpdateEmailBody): Observable<BasicResponse>

    @POST("api/users/update-email-code/")
    fun sendCodeForUpdateEmail(@Body updateEmailCodeBody: UpdateEmailCodeBody): Observable<BasicResponse>

    @POST("api/users/forgot-password")
    fun sendForgotPasswordCode(@Body sendForgotPasswordCodeBody: SendForgotPasswordCodeBody): Observable<BasicResponse>

    @POST("api/users/confirm-forgot-password/")
    fun confirmCodeForForgotPassword(@Body confirmCodeForForgotPasswordBody: ConfirmCodeForForgotPasswordBody): Observable<BasicResponse>

    @GET("api/fleet")
    fun getUserFleet(): Observable<GetUserFleetsResponse>

    @PUT("api/users/delete")
    fun deleteAccount():Observable<BasicResponse>
}